underfs/swift/src/main/java/alluxio/underfs/swift/KeystoneV3AccessProvider.java

```diff
@@ -71,61 +73,66 @@ public Access authenticate() {
         return null;
       }
 
-      HttpURLConnection connection = null;
-      BufferedReader bufReader = null;
+      CloseableHttpClient client = HttpClients.createDefault();
       try {
         // Send request
-        connection = (HttpURLConnection) new URL(mAccountConfig.getAuthUrl()).openConnection();
-        connection.setDoOutput(true);
-        connection.setRequestProperty("Accept", "application/json");
-        connection.setRequestProperty("Content-Type", "application/json");
-        OutputStream output = connection.getOutputStream();
-        output.write(requestBody.toString().getBytes());
-
-        // Parse response
-        if (connection.getResponseCode() != RESPONSE_OK) {
-          return null;
-        }
-        String token = connection.getHeaderField("X-Subject-Token");
-
-        // Parse response body
-        bufReader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
-        String responseBody = bufReader.readLine();
-        KeystoneV3Response response;
+        HttpPost post = new HttpPost(mAccountConfig.getAuthUrl());
+        post.addHeader("Accept", "application/json");
+        post.addHeader("Content-Type", "application/json");
+        post.setEntity(new ByteArrayEntity(requestBody.toString().getBytes()));
+        CloseableHttpResponse httpResponse = client.execute(post);
         try {
-          response = new ObjectMapper().readerFor(KeystoneV3Response.class).readValue(responseBody);
-        } catch (JsonProcessingException e) {
-          LOG.error("Error processing JSON response: {}", e.getMessage());
-          return null;
-        }
-        // Find endpoints
-        String internalURL = null;
-        String publicURL = null;
-        for (Catalog catalog : response.mToken.mCatalog) {
-          if (catalog.mName.equals("swift") && catalog.mType.equals("object-store")) {
-            for (Endpoint endpoint : catalog.mEndpoints) {
-              if (endpoint.mRegion.equals(mAccountConfig.getPreferredRegion())) {
-                if (endpoint.mInterface.equals("public")) {
-                  publicURL = endpoint.mUrl;
-                } else if (endpoint.mInterface.equals("internal")) {
-                  internalURL = endpoint.mUrl;
+
+          // Parse response
+          int responseCode = httpResponse.getStatusLine().getStatusCode();
+          if (responseCode != RESPONSE_OK) {
+            LOG.error("Error with response code {} ", responseCode);
+            return null;
+          }
+          String token = httpResponse.getFirstHeader("X-Subject-Token").getValue();
+
+          // Parse response body
+          BufferedReader bufReader =
+              new BufferedReader(new InputStreamReader(httpResponse.getEntity().getContent()));
```

Comment:
@ifcharming Closing `BufferedReader` will trigger closing the entity input stream. Is there a benefit of using `EntityUtils`? (looked like it will do the same thing)